### PR TITLE
Relax unsqueeze

### PR DESCRIPTION
**Pull request by swimdi**
### Ticket
#386 

### Problem description
I was going to support unsqueeze guided by [here](https://github.com/tenstorrent/pytorch2.0_ttnn/issues/386#issuecomment-2479097908), but I found latest tt-metal already support all input variation except zero element num, so I just relax the unsqueeze lowering constraint and it's now fully lowered in llama 

### What's changed
 - Relax lowering constraint of unsqueeze
 - Add guard of GPTNeo

### CODE REVIEW

File: torch_ttnn/passes/lowering/to_tt_pass.py [hunk 1]
```diff
@@ -672,17 +672,11 @@ def rewrite_node(node):
...
-                else:
-                    input_size = args[0].meta["val"].size()
-
-                output_size = node.meta["val"].size()
-                output_size = list(output_size)
-                if output_size[-1] == input_size[-1]:
-                    return g.call_function(ttnn.reshape, args=(args[0], output_size))
-                return None
+                output_shape_num_element = node.meta["val"].numel()
+                if output_shape_num_element == 0:
```
Comment (ayerofieiev-tt) [inline torch_ttnn/passes/lowering/to_tt_pass.py:676]: Can you please open a PR into metal with this check in the Op itself? 
Good to leave this check here for now too, but lets move to tt-metal

File: torch_ttnn/passes/lowering/to_tt_guard.py [hunk 2]
```diff
@@ -214,11 +214,24 @@
...
+
+# ERROR tests/models/gpt_neo/test_gpt_neo.py::test_gpt_neo[eval] -
+# RuntimeError: probability tensor contains either `inf`, `nan` or element < 0
+
 ############################################################
 
 GUARD[torch.ops.aten.add.Tensor] = partial(guard_aten, aten_add_Tensor_blocklist)
 GUARD[torch.ops.aten._adaptive_avg_pool2d.default] = partial(guard_aten, aten__adaptive_avg_pool2d_default_blocklist)
 GUARD[torch.ops.aten.view.default] = partial(guard_aten, aten_view_default_blocklist)
+GUARD[torch.ops.aten.select.int] = partial(guard_aten, aten_select_int_blocklist)
```
Comment (ayerofieiev-tt) [inline torch_ttnn/passes/lowering/to_tt_guard.py:234]: not a part of this PR, right?

**Review by ayerofieiev-tt — APPROVED** [review]

File: torch_ttnn/passes/lowering/to_tt_guard.py (hunk 2 above)
Comment (swimdi) [reply torch_ttnn/passes/lowering/to_tt_guard.py:234]: it's part of guard function of GPTNeo, and I’d like to include it here as well

File: torch_ttnn/passes/lowering/to_tt_pass.py (hunk 1 above)
Comment (swimdi) [reply torch_ttnn/passes/lowering/to_tt_pass.py:676]: Ok, I'll add this check at tt-metal

Comment (swimdi) [reply torch_ttnn/passes/lowering/to_tt_pass.py:676]: I open in there
https://github.com/tenstorrent/tt-metal/pull/15289

**Pull request merged**